### PR TITLE
Sleep minor revisions

### PR DESCRIPTION
**Pull request by jhmigueles**
This contains some minor revisions of code related to the detection of the sleep period time. These are the minor bugs I found while testing the HorAngle algorithm. There is no issue related, just let me know if you want me to create the issue for these.

- **Detection of daylight saving time days improved:** I found that partial last days, those recording finishing between 11:00 and 11:59 were mistaken as daylight saving time days, and then the end of the SPT was shifted one hour.
- **Fraction_night_invalid** is now re-calculated for the 6pm-6pm windows in the days classified as "daysleeper".

### Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Clean code has been attempted, e.g. intuitive object names and no code redundancy.
- [ ] Documentation updated:
  - [ ] Function documentation
  - [ ] Chapter vignettes for GitHub IO
  - [ ] Vignettes for CRAN
- [ ] Corresponding issue tagged in PR message. If no issue exist, please create an issue and tag it.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.
- [ ] GGIR parameters were added/removed. If yes, please also complete checklist below.

**If NEW GGIR parameter(s) were added then these NEW parameter(s) are:**
- [ ] documented in `man/GGIR.Rd`
- [ ] included with a default in `R/load_params.R`
- [ ] included with value class check in `R/check_params.R`
- [ ] included in table of `vignettes/GGIRParameters.Rmd` with references to the GGIR parts the parameter is used in.
- [ ] mentioned in NEWS.Rd as NEW parameter

**If GGIR parameter(s) were deprecated these parameter(s) are:**
- [ ] documented as deprecated in `man/GGIR.Rd`
- [ ] removed from `R/load_params.R`
- [ ] removed from `R/check_params.R`
- [ ] removed from table in `vignettes/GGIRParameters.Rmd`
- [ ] mentioned as deprecated parameter in NEWS.Rd
- [ ] added to the list in `R/extract_params.R` with deprecated parameters such that these do not produce warnings when found in old config.csv files.

### CODE REVIEW

File: R/g.sib.det.R [hunk 1]
```diff
@@ -26,7 +26,10 @@ g.sib.det = function(M, IMP, I, twd = c(-12, 12),
     time_SPTE_end_hr = as.numeric(format(time_SPTE_end, format = "%H"))
     time_SPTE_start_hr = as.numeric(format(time_SPTE_start, format = "%H"))
     t0 = as.numeric(format(iso8601chartime2POSIX(tmpTIME[1], tz = tz), format = "%H"))
-    t1 = as.numeric(format(iso8601chartime2POSIX(tmpTIME[length(tmpTIME)], tz = tz), format = "%H"))
+    # redefinition of t1 in case window ends some epochs before 12:00:00
+    t1hms = format(iso8601chartime2POSIX(tmpTIME[length(tmpTIME)], tz = tz), format = "%H:%M:%S")
+    t1hms = as.numeric(unlist(strsplit(t1hms, ":", fixed = T)))
+    t1 = ceiling(sum(c(t1hms[1], t1hms[2]/60, t1hms[3]/3600)))
```
Comment (vincentvanhees) [inline R/g.sib.det.R:32]: I am trying to understanding whether this change affect the functionality of the code.

The main objective of this section is to identify whether there was DST or not in the noon-noon window. We do this by comparing time passed based on number of epochs and time passed based on difference in timestamps.
- If delta time from timestamps > delta time from epochs + 0.1 then clock moved forward.
- If delta time from timestamps < delta time from epochs + 0.1 then clock moved backward.
I am not sure now why I used 0.1.

Can you confirm that this still works and DST is correctly handled?

File: R/g.sib.det.R [hunk 2]
```diff
@@ -310,8 +313,18 @@ g.sib.det = function(M, IMP, I, twd = c(-12, 12),
             if (newqqq2 > length(anglez)) newqqq2 = length(anglez)
             # only try to extract SPT again if it is possible to extract a window of more than 23 hour
             if (newqqq2 < length(anglez) & (newqqq2 - newqqq1) > (23*(3600/ws3)) ) {
+              # Recalculate nonwear percentage for new window (6pm to 6pm)
+              nonwear_percentage = (length(which(invalid[newqqq1:newqqq2] == 1)) /  (newqqq2 - newqqq1 + 1)) * 100
+              guider_to_use = 1
+              if (params_sleep[["HASPT.algo"]][1] == "NotWorn" &&
+                  nonwear_percentage < 25 &&
+                  length(params_sleep[["HASPT.algo"]]) == 2) {
```
Comment (vincentvanhees) [inline R/g.sib.det.R:321]: This if-statement is the same as the if statement earlier in this function https://github.com/wadpac/GGIR/blob/master/R/g.sib.det.R#L253-L256, so in the future we will have to make sure that they stay the same. 

How about moving them to a local function and calling the function in both places? Another option is to add a comment in both places to remind ourselves that the condition occurs twice in this function and needs to remain the same.

File: NEWS.md [hunk 3]
```diff
@@ -1,8 +1,25 @@
+# CHANGES IN GGIR VERSION 3.1-?
+
+- Part 4: 
+
+  - Identification of daylight saving time days in the detection of the spt is improved as it had the risk to misclassify partial first/last days as daylight saving time.
```
Comment (vincentvanhees) [inline NEWS.md:5]: This only affects the last day, correct? If it also affects the first day shouldn't object `t0` be extracted with minutes and seconds too?

File: NEWS.md [hunk 4]
```diff
@@ -1,8 +1,25 @@
+# CHANGES IN GGIR VERSION 3.1-?
+
+- Part 4:
```
Comment (vincentvanhees) [inline NEWS.md:3]: These are part 3 functions.

File: NEWS.md [hunk 5]
```diff
@@ -1,8 +1,25 @@
+# CHANGES IN GGIR VERSION 3.1-?
+
+- Part 4: 
+
+  - Identification of daylight saving time days in the detection of the spt is improved as it had the risk to misclassify partial first/last days as daylight saving time.
+  
+  - In days classified as daysleeper, the window over which the fraction_night_invalid is calculated now also shifts to 6pm-6pm, as it used to report the nonwear within 12pm-12pm.
+  
+
 # CHANGES IN GGIR VERSION 3.1-8
```
Comment (vincentvanhees) [inline NEWS.md:10]: It seems 3.1-8 items overlap with 3.1-9 items.

**Review by vincentvanhees — CHANGES_REQUESTED** [review]
Thanks Jairo, please find my comments below. I am happy to help work on this, but maybe easier if you could first have a look at it.

File: R/g.sib.det.R (hunk 1 above)
Comment (jhmigueles) [reply R/g.sib.det.R:32]: I am now investigating this, and the error in the master branch only appears when the last timestamp is 11:59:55, then the day is considered to be a DST day even if it is not. Also, the approach only adjusts the `SPT_end` time, but not the `SPT_start` time (if sleep onset occurs after 2am, then the start time also needs to be adjusted).

In my approach in this branch, the issue with 11:59:55 is correctly handled, but I have found other errors (it fails when the last timestamp is earlier than 11:59:55), so I need to work more on this.

I will reverse the changes here and try to find a way to solve the original issue without introducing other bugs... I am thinking about getting the timezone specification (`"%z"`) from the ISO8601 format for the first timestamp (e.g., `"+0100"`), and get the same for the spt_start and spt_end timestamps. If `%z` is different between `t0` and `SPT_start `and/or `SPT_end`, then the time would need to be adjusted.

Comment (jhmigueles) [reply R/g.sib.det.R]: Now this simplified option is implemented and it has been tested in different scenarios.

**Review by vincentvanhees — APPROVED** [review]

**Pull request merged**